### PR TITLE
refactor: create cli module

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Create a share CLI module

Related to: #145

### How I did it

Discussed here: #140  on how to refactor the CLI to re-use shared options and utils, both for us and for 3rd party plugins.

### How to verify it

Make sure you can import the shared CLI definitions

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: src/ape/_cli.py [hunk 1]
```diff
@@ -5,9 +5,9 @@
 import click
 import yaml
 
+from ape.cli.utils import Abort, notify
```
Comment (fubuloubu) [inline src/ape/_cli.py:8]: We can probably just export these at the top level
```suggestion
from ape.cli import Abort, notify
```

File: src/ape_run/_cli.py [hunk 2]
```diff
@@ -5,8 +5,9 @@
 import click
 
 from ape import config, networks
-from ape.options import network_option, verbose_option
-from ape.utils import Abort, get_relative_path
+from ape.cli.options import network_option, verbose_option
```
Comment (fubuloubu) [inline src/ape_run/_cli.py:8]: Same here
```suggestion
from ape.cli import network_option, verbose_option
```

File: src/ape/_cli.py (hunk 1 above)
Comment (unparalleled-js) [reply src/ape/_cli.py:8]: This creates a bit of stir, regarding circular imports and stuff..
I am going to try and dive and deeper into the structure and see what might be best.

**Comment by unparalleled-js** [comment]
closing until logging work is done

**Pull request closed without merging**